pallets/portfolio/src/lib.rs

```diff
@@ -0,0 +1,261 @@
+                    did,
+                    Some(num),
+                    None,
+                    ticker,
+                    balance,
+                ));
+            }
+            <PortfolioAssetBalances<T>>::remove_prefix(&portfolio_id);
+            <Portfolios>::remove(&did, &num);
+            Self::deposit_event(RawEvent::PortfolioDeleted(did, num));
+            Ok(())
+        }
+
+        /// Moves a token amount from one portfolio of an identity to another portfolio of the same
+        /// identity.
+        #[weight = 250_000 + 250_000 * u64::try_from(items.len()).unwrap_or_default()]
+        pub fn move_portfolio(
+            origin,
+            from_num: Option<PortfolioNumber>,
+            to_num: Option<PortfolioNumber>,
+            items: Vec<MovePortfolioItem<<T as CommonTrait>::Balance>>,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let did = Context::current_identity_or::<Identity<T>>(&sender)?;
+            ensure!(from_num != to_num, Error::<T>::DestinationIsSamePortfolio);
+            if let Some(from_num) = from_num {
+                ensure!(Self::portfolios(&did, from_num).is_some(), Error::<T>::PortfolioDoesNotExist);
+            }
+            if let Some(to_num) = to_num {
+                ensure!(Self::portfolios(&did, to_num).is_some(), Error::<T>::PortfolioDoesNotExist);
+            }
+            let from_portfolio_id = from_num
+                .and_then(|num| Some(PortfolioId::user_portfolio(did, num)))
+                .unwrap_or_else(|| PortfolioId::default_portfolio(did));
+            let to_portfolio_id = to_num
+                .and_then(|num| Some(PortfolioId::user_portfolio(did, num)))
+                .unwrap_or_else(|| PortfolioId::default_portfolio(did));
+            for item in items {
+                let from_balance = Self::portfolio_asset_balances(&from_portfolio_id, &item.ticker);
+                ensure!(from_balance >= item.amount, Error::<T>::InsufficientPortfolioBalance);
+                <PortfolioAssetBalances<T>>::insert(
+                    &from_portfolio_id,
+                    &item.ticker,
+                    from_balance - item.amount
+                );
+                let to_balance = Self::portfolio_asset_balances(&to_portfolio_id, &item.ticker);
+                <PortfolioAssetBalances<T>>::insert(
+                    &to_portfolio_id,
+                    &item.ticker,
+                    to_balance.saturating_add(item.amount)
+                );
+                Self::deposit_event(RawEvent::MovedBetweenPortfolios(
+                    did,
+                    from_num,
+                    to_num,
+                    item.ticker,
+                    item.amount
+                ));
+            }
+            Ok(())
+        }
+
+        /// Renames a non-default portfolio.
+        #[weight = 500_000]
+        pub fn rename_portfolio(
```

Comment:
Only the two first lines are the same and then the functions diverge. This doesn't look like a couple of good candidates. Suggest how you think they should be deduplicated or we can leave them as is because we normally don't deduplicate those two initial lines.

Reply:
Personally, I would deduplicate the `let name_uniq` and the subsequent `ensure!`; that logic is at least common.

Reply:
I wouldn't move `name_uniq` earlier in `rename_portfolio` because checking portfolio existence should be done first.

`name_uniq` checks do different things in those two functions:
1. Check that the new portfolio can be created called `name`.
2. Check that an existing portfolio can be renamed as `name`.

The second check depends on existence of that portfolio that should be checked first. I hope that clarifies why these are really not the best candidates for such refactoring.

Reply:
For this comment I implemented a uniqueness check in a `fn` which is called by the two extrinsics in question.